[3.0.0-RC1] PS256 is not supported while building with old JDK8

I am getting the following error when building using an old version of Java8 (JDK 1.8.0_112)

@sberyozkin has indicated [here](https://github.com/eclipse/microprofile-jwt-auth/issues/227#issuecomment-742862081) that PS256 is not a required algorithm to support for MP-JWT 1.2 and only RS256, ES256 are required signature algorithms.

```
[ERROR] Tests run: 24, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.274 s <<< FAILURE! - in io.smallrye.jwt.build.JwtSignTest
[ERROR] testSignClaimsPS256(io.smallrye.jwt.build.JwtSignTest)  Time elapsed: 0.008 s  <<< ERROR!
io.smallrye.jwt.build.JwtSignatureException: SRJWT05012: Failure to create a signed JWT token: PS256 is an unknown, unsupported or unavailable alg algorithm (not one of [none, HS256, HS384, HS512, ES256, ES384, ES512, RS256, RS384, RS512]).
        at io.smallrye.jwt.build.JwtSignTest.testSignClaimsPS256(JwtSignTest.java:431)
Caused by: org.jose4j.lang.InvalidAlgorithmException: PS256 is an unknown, unsupported or unavailable alg algorithm (not one of [none, HS256, HS384, HS512, ES256, ES384, ES512, RS256, RS384, RS512]).
        at io.smallrye.jwt.build.JwtSignTest.testSignClaimsPS256(JwtSignTest.java:431)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   JwtSignTest.testSignClaimsPS256:431 » JwtSignature SRJWT05012: Failure to crea...
[INFO]
[ERROR] Tests run: 58, Failures: 0, Errors: 1, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for SmallRye: MicroProfile JWT Parent 3.0.0-RC1:
[INFO]
[INFO] SmallRye: MicroProfile JWT Parent .................. SUCCESS [  8.091 s]
[INFO] SmallRye: MicroProfile JWT - Implementation Parent . SUCCESS [  1.384 s]
[INFO] SmallRye: MicroProfile JWT Implementation Common ... SUCCESS [  8.242 s]
[INFO] SmallRye: MicroProfile JWT Build Implementation .... FAILURE [ 10.138 s]
[INFO] SmallRye: MicroProfile JWT Implementation .......... SKIPPED
[INFO] SmallRye: MicroProfile JWT - Testsuite Parent ...... SKIPPED
[INFO] SmallRye: MicroProfile JWT Basic Testsuite ......... SKIPPED
[INFO] SmallRye: MicroProfile JWT TCK ..................... SKIPPED
[INFO] Empty Release Project to Avoid Maven Bug ........... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

--- a/implementation/jwt-build/pom.xml
+++ b/implementation/jwt-build/pom.xml
@@ -84,6 +84,11 @@
       <artifactId>smallrye-config</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcprov-jdk15on</artifactId>
+      <scope>test</scope>
+    </dependency>
   </dependencies>
 
   <build>

--- a/implementation/jwt-build/src/test/java/io/smallrye/jwt/build/JwtSignPS256Test.java
+++ b/implementation/jwt-build/src/test/java/io/smallrye/jwt/build/JwtSignPS256Test.java
@@ -1,0 +1,73 @@
+/*
+ *   Copyright 2019 Red Hat, Inc, and individual contributors.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package io.smallrye.jwt.build;
+
+import java.security.Security;
+
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.lang.BouncyCastleProviderHelp;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.smallrye.jwt.algorithm.SignatureAlgorithm;
+// import io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipal;
+import io.smallrye.jwt.util.KeyUtils;
+
+public class JwtSignPS256Test {
+    @BeforeClass
+    public static void installBouncyCastleProviderIfNeeded() {
+        if (!isPS256Supported()) {
+            BouncyCastleProviderHelp.enableBouncyCastleProvider();
+        }
+    }
+
+    @AfterClass
+    public static void uninstallBouncyCastleProviderIfNeeded() {
+        if (!isPS256Supported()) {
+            Security.removeProvider("org.bouncycastle.jce.provider.BouncyCastleProvider");
+        }
+    }
+
+    private static boolean isPS256Supported() {
+        for (String sigAlg : Security.getAlgorithms("Signature")) {
+            if ("RSASSA-PSS".equalsIgnoreCase(sigAlg)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Test
+    public void testSignClaimsPS256() throws Exception {
+        String jwt = Jwt.claims()
+                .claim("customClaim", "custom-value")
+                .jws().algorithm(SignatureAlgorithm.PS256)
+                .sign("/privateKey.pem");
+
+        JsonWebSignature jws = JwtSignTest.getVerifiedJws(jwt, KeyUtils.readPublicKey("/publicKey.pem"));
+        JwtClaims claims = JwtClaims.parse(jws.getPayload());
+
+        Assert.assertEquals(4, claims.getClaimsMap().size());
+        JwtSignTest.checkDefaultClaimsAndHeaders(JwtSignTest.getJwsHeaders(jwt, 2), claims, "PS256", 300);
+
+        Assert.assertEquals("custom-value", claims.getClaimValue("customClaim"));
+    }
+
+}

--- a/implementation/jwt-build/src/test/java/io/smallrye/jwt/build/JwtSignTest.java
+++ b/implementation/jwt-build/src/test/java/io/smallrye/jwt/build/JwtSignTest.java
@@ -53,7 +53,6 @@ import org.jose4j.keys.EllipticCurves;
 import org.junit.Assert;
 import org.junit.Test;
 
-import io.smallrye.jwt.algorithm.SignatureAlgorithm;
 // import io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipal;
 import io.smallrye.jwt.util.KeyUtils;
 
@@ -286,7 +285,7 @@ public class JwtSignTest {
         return getVerifiedJws(jwt, getPublicKey());
     }
 
-    private static JsonWebSignature getVerifiedJws(String jwt, Key key) throws Exception {
+    static JsonWebSignature getVerifiedJws(String jwt, Key key) throws Exception {
         JsonWebSignature jws = new JsonWebSignature();
         jws.setKey(key);
         jws.setCompactSerialization(jwt);
@@ -298,7 +297,7 @@ public class JwtSignTest {
         checkDefaultClaimsAndHeaders(headers, claims, "RS256", 300);
     }
 
-    private static void checkDefaultClaimsAndHeaders(Map<String, Object> headers, JwtClaims claims, String algo,
+    static void checkDefaultClaimsAndHeaders(Map<String, Object> headers, JwtClaims claims, String algo,
             long expectedLifespan)
             throws Exception {
         NumericDate iat = claims.getIssuedAt();
@@ -423,22 +422,6 @@ public class JwtSignTest {
         Assert.assertEquals("custom-value", claims.getClaimValue("customClaim"));
     }
 
-    @Test
-    public void testSignClaimsPS256() throws Exception {
-        String jwt = Jwt.claims()
-                .claim("customClaim", "custom-value")
-                .jws().algorithm(SignatureAlgorithm.PS256)
-                .sign();
-
-        JsonWebSignature jws = getVerifiedJws(jwt, getPublicKey());
-        JwtClaims claims = JwtClaims.parse(jws.getPayload());
-
-        Assert.assertEquals(4, claims.getClaimsMap().size());
-        checkDefaultClaimsAndHeaders(getJwsHeaders(jwt, 2), claims, "PS256", 300);
-
-        Assert.assertEquals("custom-value", claims.getClaimValue("customClaim"));
-    }
-
     private static EllipticCurveJsonWebKey createECJwk() throws Exception {
         return EcJwkGenerator.generateJwk(EllipticCurves.P256);
     }
@@ -536,7 +519,7 @@ public class JwtSignTest {
         }
     }
 
-    private static Map<String, Object> getJwsHeaders(String compactJws, int expectedSize) throws Exception {
+    static Map<String, Object> getJwsHeaders(String compactJws, int expectedSize) throws Exception {
         int firstDot = compactJws.indexOf(".");
         String headersJson = new Base64Url().base64UrlDecodeToUtf8String(compactJws.substring(0, firstDot));
         Map<String, Object> headers = JsonUtil.parseJson(headersJson);

--- a/pom.xml
+++ b/pom.xml
@@ -33,6 +33,7 @@
   <url>http://smallrye.io</url>
 
   <properties>
+    <bouncycastle.version>1.67</bouncycastle.version>
     <version.jakarta.servlet.api>4.0.4</version.jakarta.servlet.api>
     <version.jakarta.security.enterprise.api>1.0.2</version.jakarta.security.enterprise.api>
     <version.eclipse.microprofile.jwt>1.1.1</version.eclipse.microprofile.jwt>
@@ -148,6 +149,11 @@
         <groupId>io.smallrye.config</groupId>
         <artifactId>smallrye-config</artifactId>
         <version>${version.io.smallrye.config}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.bouncycastle</groupId>
+        <artifactId>bcprov-jdk15on</artifactId>
+        <version>${bouncycastle.version}</version>
       </dependency>
     </dependencies>
   </dependencyManagement>

